WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -201,11 +203,20 @@ def process_payment
       registrations = competition.registrations
       registration = registrations.find_by_user_id!(current_user.id)
     end
-    token = params[:stripeToken]
+    token, amount_param = params.require(:payment).require([:stripeToken, :total_amount])
+    amount = amount_param.to_i
+
+    # 'amount' has not been check by anyone, and could be user-crafted; validate it!
+    # if 'token' is wrong, it's Stripe which will complain
+    if amount < registration.outstanding_entry_fees.cents
```

Comment:
Just checking: are we okay with overpaying because of donations? If so, would it make sense to make that more explicit? (this does not need to hold up the PR)